### ISSUE TITLE
Equalizer and other filters

### ISSUE THREAD

**Issue by Harvie** [PipeWire/pipewire#109]
Will it be possible to have equalizer (and similar filters) in pipewire? Some of my speakers need sound to be preprocessed a bit to give decent listening quality and i don't have hardware equalizer.

**Comment by rektide** [comment]
I'd hoped that the jack-compatibility in pipewire might allow using jack filters to effect such changes, but I'm not sure whether anyone else had that use case (of jack being an intermediary part of an audio pipeline) in mind.

**Comment by alexpolevoy** [comment]
Why would you ever need an equalizer? You don't need to EQ anything, it ruins the sound if used not in a professional environment by untrained personnel. A better improvement to your listening experience would be replacing your speakers with something more decent. No sort of an EQ can improve your speakers' tonal balance.

**Comment by Hi-Angel** [comment]
@alexpolevoy for example, different headphones often accentuate different tones. So you may need to tune sound with equalizer to improve it to your liking.

**Comment by RafaelLinux** [comment]
I agree. I think a equalizer implemented on PipeWire is necessary. User don't need to be a professional to choose between presets that adjust to his current audio output (external speakers, headphones, monitor speakers, 5.1 surround, etc). It's possible many years ago in Windows, with the software that comes with the motherboard or sound card and even Windows 10 has audio presets. Each time I try the same with equalizers that comes with audio players like Audacious, QMMP, Clementine, etc finally I have had a terrible result, with saturated sound even lowering the "preamplification". So, finally, it's a feature unusable. That  issue does not happen in Windows, if you know how to adjust preamplification and each frequency slider or use the presets. And it's logical to have a common equalizer for all audio streaming, cause if for example you are using headphones, you want to hear the sound with the "headphones" preset (if available) or the own user audio preset, according with his preferences and hardware. 

Regards

**Comment by Harvie** [comment]
BTW i think such equalizer should have global settings and per-output settings. So i can have different EQ on each soundcard. Also laptops nowadays have headphone jack detection available in software, so it should be possible to switch between presets when you insert headphones.

Pulse audio allows you to have different volume preset on single soundcard for headphones and internal speakers, so when i plug headphones into my laptop, it automaticaly changes the volume. I think equalizer should work the same (unless you explicitly disable per-output eq).

**Comment by RafaelLinux** [comment]
I agree!!! However, talking of PulseAudio, at least with Plasma default configuration in OpenSUSE, the presets autoload on depending if I plug headphones or using speakers,  is not disposable. Maybe it's necessary to install additional packages.

**Comment by wtay** [comment]
We have not exactly worked out what we're going to do here. This is all a policy of the session-manager. PipeWire allows you to insert all the ladspa/lv2 plugins you want with javl or jack-rack and so. W're probably going to insert effects before the sinks (configurable) And then the session manager connects you to those when needed.
